test/unit/setting_registry_test.rb

```diff
@@ -126,6 +126,29 @@ class SettingRegistryTest < ActiveSupport::TestCase
       end
     end
 
+    context 'string setting' do
+      setup do
+        registry._add('test',
+          category: 'Setting',
+          default: nil,
+          type: :string,
+          full_name: 'test Foo',
+          description: 'test update',
+          encrypted: false,
+          context: :test)
+      end
+
+      it 'does not distinguish between nil default and an empty string' do
+        Setting.any_instance.stubs(:setting_definition).returns(registry.find('test'))
+        registry.set_user_value('test', 'foobar').save!
+        s = registry.set_user_value('test', '')
+
+        assert s.save
+        assert_nil s.value
+        assert_nil s.read_attribute(:value)
```

Comment:
For my understanding, why this in addition to `s.value`?

Reply:
It is a bit of a stretch, but in theory it could happen that `s.value` would be `nil`, but the db field would be a nil serialized to yaml (`---\n`) instead of a proper null